src/relay/op/tensor/nn.cc

```diff
@@ -0,0 +1,104 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file nn.cc
+ * \brief Property def of nn operators.
+ */
+#include <tvm/relay/expr.h>
+#include <tvm/relay/op.h>
+#include <tvm/relay/attrs/nn.h>
+#include "../type_relations.h"
+
+namespace tvm {
+namespace relay {
+
+TVM_REGISTER_NODE_TYPE(DenseAttrs);
+
+
+bool DenseRel(const Array<Type>& types,
+              int num_inputs,
+              const Attrs& attrs,
+              const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 3);
+  const auto* data = types[0].as<TensorTypeNode>();
+  if (data == nullptr) return false;
+  const auto* weight = types[1].as<TensorTypeNode>();
+  if (weight == nullptr) return false;
+
+  const DenseAttrs* param = attrs.as<DenseAttrs>();
+  CHECK(param != nullptr);
+
+  CHECK(static_cast<int>(data->shape.size()) != 0);
+  Array<tvm::Expr> wshape = weight->shape;
+
+  if (param->units != 0) {
+    // CHECK_EQ(param->units == wshape[wshape.size() - 1])
```

Comment:
i want to do this comparison (Integer and Expr).. @srkreddy1238 @tqchen How can i do this comparison?

Reply:
do CHECK(reporter->AssertEQ(param->units, wshape[wshape.size()-1]))